src/Plugins/ComplexCore/test/CombineAttributeArraysTest.cpp

```diff
@@ -0,0 +1,342 @@
+  for(const auto& inputDataPath : inputDataPaths)
+  {
+    auto& inputArray = dataStructure.template getDataRefAs<DataArrayType>(inputDataPath);
+    size_t numInputComps = inputArray.getNumberOfComponents();
+    for(size_t tupleIndex = 0; tupleIndex < numTuples; tupleIndex++)
+    {
+      for(size_t compIndex = 0; compIndex < numInputComps; compIndex++)
+      {
+        REQUIRE(output[tupleIndex * numOutputComps + compIndex + compOffset] == inputArray[tupleIndex * numInputComps + compIndex]);
+      }
+    }
+    compOffset += numInputComps;
+  }
+
+  return {};
+}
+
+TEST_CASE("Core::CombineAttributeArrays: Algorithm Validation", "[Core][CombineAttributeArrays]")
+{
+  MultiArraySelectionParameter::ValueType inputArrays = {DataPath({k_Array1}), DataPath({k_Array2}), DataPath({k_Array3})};
+
+  SECTION("UINT8")
+  {
+    DataStructure dataStructure = CreateTestDataStructure<uint8_t>();
+    Arguments args;
+    CombineAttributeArraysFilter filter;
+
+    // Create default Parameters for the filter.
+    args.insertOrAssign(CombineAttributeArraysFilter::k_NormalizeData_Key, std::make_any<bool>(false));
+    args.insertOrAssign(CombineAttributeArraysFilter::k_MoveValues_Key, std::make_any<bool>(false));
+    args.insertOrAssign(CombineAttributeArraysFilter::k_SelectedDataArrayPaths_Key, std::make_any<MultiArraySelectionParameter::ValueType>(inputArrays));
+    args.insertOrAssign(CombineAttributeArraysFilter::k_StackedDataArrayName_Key, std::make_any<DataObjectNameParameter::ValueType>(k_OutputArray));
+
+    // Preflight the filter and check result
+    auto preflightResult = filter.preflight(dataStructure, args);
+    COMPLEX_RESULT_REQUIRE_VALID(preflightResult.outputActions);
+
+    auto result = filter.execute(dataStructure, args);
+    COMPLEX_RESULT_REQUIRE_VALID(result.result);
+
+    Result<> validationResult = ValidateFilterOutput<uint8_t>(dataStructure);
+    COMPLEX_RESULT_REQUIRE_VALID(validationResult)
+  }
+
+  SECTION("INT8")
+  {
+    DataStructure dataStructure = CreateTestDataStructure<int8_t>();
+    Arguments args;
+    CombineAttributeArraysFilter filter;
+
+    // Create default Parameters for the filter.
+    args.insertOrAssign(CombineAttributeArraysFilter::k_NormalizeData_Key, std::make_any<bool>(false));
+    args.insertOrAssign(CombineAttributeArraysFilter::k_MoveValues_Key, std::make_any<bool>(false));
+    args.insertOrAssign(CombineAttributeArraysFilter::k_SelectedDataArrayPaths_Key, std::make_any<MultiArraySelectionParameter::ValueType>(inputArrays));
+    args.insertOrAssign(CombineAttributeArraysFilter::k_StackedDataArrayName_Key, std::make_any<DataObjectNameParameter::ValueType>(k_OutputArray));
+
+    // Preflight the filter and check result
+    auto preflightResult = filter.preflight(dataStructure, args);
+    COMPLEX_RESULT_REQUIRE_VALID(preflightResult.outputActions);
+
+    auto result = filter.execute(dataStructure, args);
+    COMPLEX_RESULT_REQUIRE_VALID(result.result);
+  }
```

Comment:
@imikejackson Did you purposely not add the ValidateFilterOutput function check to the rest of these sections like you did in the UINT8 section?